msgraph-developer-proxy-plugins/Guidance/SelectGuidancePlugin.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Graph.DeveloperProxy.Abstractions;
+using System.Text.RegularExpressions;
+using Titanium.Web.Proxy.Http;
+
+namespace Microsoft.Graph.DeveloperProxy.Plugins.Guidance;
+
+public class SelectGuidancePlugin : IProxyPlugin {
+    private ISet<Regex>? _urlsToWatch;
+    private ILogger? _logger;
+    public string Name => nameof(SelectGuidancePlugin);
+
+    public void Register(IPluginEvents pluginEvents,
+                            IProxyContext context,
+                            ISet<Regex> urlsToWatch,
+                            IConfigurationSection? configSection = null) {
+        if (pluginEvents is null) {
+            throw new ArgumentNullException(nameof(pluginEvents));
+        }
+
+        if (context is null || context.Logger is null) {
+            throw new ArgumentException($"{nameof(context)} must not be null and must supply a non-null Logger", nameof(context));
+        }
+
+        if (urlsToWatch is null || urlsToWatch.Count == 0) {
+            throw new ArgumentException($"{nameof(urlsToWatch)} cannot be null or empty", nameof(urlsToWatch));
+        }
+
+        _urlsToWatch = urlsToWatch;
+        _logger = context.Logger;
+
+        pluginEvents.BeforeRequest += OnRequest;
```

Comment:
Wouldn't it better for this plugin to run on the AfterResponse event so that we show the guidance even if the response has been mocked?